OBSDOCS-1640: Document HTTP output proxy should be configurable (#89407)
openshift/openshift-docs: modules/log6x-cluster-logging-collector-log-forward-syslog.adoc

.Prerequisites

* You must have a logging server that is configured to receive the logging data using the specified protocol or format.

.Procedure

. Create or edit a YAML file that defines the `ClusterLogForwarder` CR object:
+
[source,yaml]
----
apiVersion: observability.openshift.io/v1
kind: ClusterLogForwarder
metadata:
  name: collector
spec:
  managementState: Managed
  outputs:
  - name: rsyslog-east # <1>
    syslog:
      appName: <app_name> # <2>
      enrichment: KubernetesMinimal
      facility: <facility_value> # <3>
      msgId: <message_ID> # <4>
      payloadKey: <record_field> # <5>
      procId: <process_ID> # <6>
      rfc: <RFC3164_or_RFC5424> # <7>
      severity: informational # <8>
      tuning:
        deliveryMode: <AtLeastOnce_or_AtMostOnce> # <9>
      url: <url> # <10>
    tls: # <11>
      ca:
        key: ca-bundle.crt
        secretName: syslog-secret
    type: syslog
  pipelines:
  - inputRefs: # <12>
    - application
    name: syslog-east # <13>
    outputRefs:
    - rsyslog-east
  serviceAccount: # <14>
    name: logcollector
----
<1> Specify a name for the output.

> <2> Optional: Specify the value for the "APP-NAME" part of the syslog message header. The value must conform with link:https://datatracker.ietf.org/doc/html/rfc5424[The Syslog Protocol]. The value can be a combination of static and dynamic values consisting of field paths followed by `||`, and then followed by another field path or a static value. The maximum length of the final values is truncated to 48 characters. You must encase a dynamic value curly brackets and the value must be followed with a static fallback value separated with `||`. Static values can only contain alphanumeric characters along with dashes, underscores, dots and forward slashes. Example value: <value1>-{.<value2>||"none"}.

Reviewer: ```suggestion
<2> Optional: Specify the value for the `appName` part of the syslog message header. The value must conform with link:https://datatracker.ietf.org/doc/html/rfc5424[The Syslog Protocol]. The value can be a combination of static and dynamic values consisting of field paths followed by `||`, and then followed by another field path or a static value. The maximum length of the final values is truncated to 48 characters. You must encase a dynamic value curly brackets and the value must be followed with a static fallback value separated with `||`. Static values can only contain alphanumeric characters along with dashes, underscores, dots and forward slashes. Example value: <value1>-{.<value2>||"none"}.
```

Reviewer: Whenever you reference inline parameters, you should use backticks: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference

> <3> Optional: Specify the value for "Facility" part of the MSGID part of the syslog-msg header.

Reviewer: ```suggestion
<3> Optional: Specify the value for `facility` part of the MSGID part of the syslog-msg header.
```

Author: updated to `Facility` and also removed extraneous "part of the MSGID" text

> <4> Optional: Specify the value for "MSGID" part of the MSGID part of the syslog-msg header. The value can be a combination of static and dynamic values consisting of field paths followed by `||`, and then followed by another field path or a static value. The maximum length of the final values is truncated to 32 characters. You must encase a dynamic value curly brackets and the value must be followed with a static fallback value separated with `||`. Static values can only contain alphanumeric characters along with dashes, underscores, dots and forward slashes. Example value: <value1>-{.<value2>||"none"}.

Reviewer: ```suggestion
<4> Optional: Specify the value for `msgId` part of the syslog-msg header. The value can be a combination of static and dynamic values consisting of field paths followed by `||`, and then followed by another field path or a static value. The maximum length of the final values is truncated to 32 characters. You must encase a dynamic value curly brackets and the value must be followed with a static fallback value separated with `||`. Static values can only contain alphanumeric characters along with dashes, underscores, dots and forward slashes. Example value: <value1>-{.<value2>||"none"}.
```

Reviewer: backticks for this and I think maybe some duplicated text?

> <5> Optional: Specify the record field to use as the payload. The PayloadKey value must be a single field path encased in single curly brackets `{}`. Example: {.<value>}.

Reviewer: ```suggestion
<5> Optional: Specify the record field to use as the payload. The `payloadKey` value must be a single field path encased in single curly brackets `{}`. Example: {.<value>}.
```

Author: fixed

> <6> Optional: Specify the value for the "PROCID" part of the syslog message header. The value must conform with link:https://datatracker.ietf.org/doc/html/rfc5424[The Syslog Protocol]. The value can be a combination of static and dynamic values consisting of field paths followed by `||`, and then followed by another field path or a static value. The maximum length of the final values is truncated to 48 characters. You must encase a dynamic value curly brackets and the value must be followed with a static fallback value separated with `||`. Static values can only contain alphanumeric characters along with dashes, underscores, dots and forward slashes. Example value: <value1>-{.<value2>||"none"}.

Reviewer: ```suggestion
<6> Optional: Specify the value for the `procId` part of the syslog message header. The value must conform with link:https://datatracker.ietf.org/doc/html/rfc5424[The Syslog Protocol]. The value can be a combination of static and dynamic values consisting of field paths followed by `||`, and then followed by another field path or a static value. The maximum length of the final values is truncated to 48 characters. You must encase a dynamic value curly brackets and the value must be followed with a static fallback value separated with `||`. Static values can only contain alphanumeric characters along with dashes, underscores, dots and forward slashes. Example value: <value1>-{.<value2>||"none"}.
```

Author: Updated as `PROCID`

<7> Optional: Set the RFC that the generated messages conform to. The value can be `RFC3164` or `RFC5424`.
<8> Optional: Set the severity level for the message. For more information, see link:https://datatracker.ietf.org/doc/html/rfc5424#section-6.2.1[The Syslog Protocol].
<9> Optional: Set the delivery mode for log forwarding. The value can be either `AtLeastOnce`, or `AtMostOnce`.
<10> Specify the absolute URL with a scheme. Valid schemes are: `tcp`, `tls`, and `udp`. For example: `tls://syslog-receiver.example.com:6514`.
<11> Specify the settings for controlling options of the transport layer security (TLS) client connections.
<12> Specify which log types to forward by using the pipeline: `application,` `infrastructure`, or `audit`.
<13> Specify a name for the pipeline.
<14> The name of your service account.

. Create the CR object:
+
[source,terminal]
----
$ oc create -f <filename>.yaml
----

[id="cluster-logging-collector-log-forward-examples-syslog-log-source_{context}"]

> == Adding log source information to message output

Reviewer: ```suggestion
== Adding log source information to the message output
```

Author: fixed

You can add `namespace_name`, `pod_name`, and `container_name` elements to the `message` field of the record by adding the `enrichment` field to your `ClusterLogForwarder` custom resource (CR).

[source,yaml]

> ----
>   spec:
>     outputs:
>     - name: syslogout
>       syslog:
>         enrichment: KubernetesMinimal: true
>         facility: user
>         payloadKey: message
>         rfc: RFC3164
>         severity: debug
>         tag: mytag
>       type: syslog
>       url: tls://syslog-receiver.example.com:6514
>     pipelines:
>     - inputRefs:
>       - application
>       name: test-app
>       outputRefs:
>       - syslogout
> ----

Reviewer: ```suggestion
----
# ...
  spec:
    outputs:
    - name: syslogout
      syslog:
        enrichment: KubernetesMinimal: true
        facility: user
        payloadKey: message
        rfc: RFC3164
        severity: debug
        tag: mytag
      type: syslog
      url: tls://syslog-receiver.example.com:6514
    pipelines:
    - inputRefs:
      - application
      name: test-app
      outputRefs:
      - syslogout
# ...      
----
```

Reviewer: I notice that the spec is spaced over, probably because there's other parameters before. I would suggest using an elipses here to make that clearer, and maybe adding the context of what comes right before that spec would be orienting to customers too?

Author: good point added "#..." at the beginning and at the end. As the example is about updating an existing CR, i don't think we need to add what comes before and after. TBH, I'll probably rewrite this with section as part of https://issues.redhat.com/browse/OBSDOCS-1747 because it is just showing what using enrichment: KubernetesMinimal: true and

Reviewer: That sounds good, @theashiot ! Thanks for updating with the ellipses and the plan to spiff this up later.

.Prerequisites

* You must have a logging server that is configured to receive the logging data using the specified protocol or format.

.Procedure

. Create or edit a YAML file that defines the `ClusterLogForwarder` CR object:
+
[source,yaml]
----
apiVersion: observability.openshift.io/v1
kind: ClusterLogForwarder
metadata:
  name: collector
spec:
  managementState: Managed
  outputs:
  - name: rsyslog-east # <1>
    syslog:
      appName: <app_name> # <2>
      enrichment: KubernetesMinimal
      facility: <facility_value> # <3>
      msgId: <message_ID> # <4>
      payloadKey: <record_field> # <5>
      procId: <process_ID> # <6>
      rfc: <RFC3164_or_RFC5424> # <7>
      severity: informational # <8>
      tuning:
        deliveryMode: <AtLeastOnce_or_AtMostOnce> # <9>
      url: <url> # <10>
    tls: # <11>
      ca:
        key: ca-bundle.crt
        secretName: syslog-secret
    type: syslog
  pipelines:
  - inputRefs: # <12>
    - application
    name: syslog-east # <13>
    outputRefs:
    - rsyslog-east
  serviceAccount: # <14>
    name: logcollector
----

> <1> Specify a name for the output.

Reviewer 2: How about add 'optional' from appName to tuning?

Author: done

<5> Specify the record field to use as the payload. The PayloadKey value must be a single field path encased in single curly brackets `{}`. Example: {.bar}.
<6> Specify the value for the "PROCID" part of the syslog message header. The value must conform with link:https://datatracker.ietf.org/doc/html/rfc5424[The Syslog Protocol]. The value can be a combination of static and dynamic values consisting of field paths followed by `||`, and then followed by another field path or a static value. The maximum length of the final values is truncated to 48 characters. You must encase a dynamic value curly brackets and the value must be followed with a static fallback value separated with `||`. Static values can only contain alphanumeric characters along with dashes, underscores, dots and forward slashes. Example value: foo-{.bar||"none"}.
<7> Set the RFC that the generated messages conform to. The value can be `RFC3164` or `RFC5424`.
<8> Set the severity level for the message. For more information, see link:https://datatracker.ietf.org/doc/html/rfc5424#section-6.2.1[The Syslog Protocol].
<9> Set the delivery mode for log forwarding. The value can be either `AtLeastOnce`, or `AtMostOnce`.

> <10> Specify the absolute URL with a scheme. Valid schemes are: `tcp`, `tls`, `udp` and `udps`. For example: `udps://syslog.example.com:1234`.

Reviewer 2: How about tls://syslog-receiver.example.com:6514

Author: updated as tls://syslog-receiver.example.com:6514

<11> Specify the settings for controlling options of the transport layer security (TLS) client connections.
<12> Specify which log types to forward by using the pipeline: `application,` `infrastructure`, or `audit`.
<13> Specify a name for the pipeline.
<14> The name of your service account.

. Create the CR object:
+
[source,terminal]
----
$ oc create -f <filename>.yaml
----

[id="cluster-logging-collector-log-forward-examples-syslog-log-source_{context}"]
== Adding log source information to message output

You can add `namespace_name`, `pod_name`, and `container_name` elements to the `message` field of the record by adding the `enrichment` field to your `ClusterLogForwarder` custom resource (CR).

[source,yaml]
----
  spec:
    outputs:
    - name: syslogout
      syslog:
        enrichment: KubernetesMinimal: true
        facility: user
        payloadKey: message
        rfc: RFC3164
        severity: debug
        tag: mytag
      type: syslog

>       url: tls://syslog-receiver.openshift-logging.svc:24224

Author: done

    pipelines:
    - inputRefs:
      - application
      name: test-app
      outputRefs:
      - syslogout
----

[NOTE]
====
This configuration is compatible with both RFC3164 and RFC5424.
====

.Example syslog message output with `enrichment: None`
[source, text]

> ----

Reviewer 2: How about remove some fileds from this example?

Author: updated, wdyt?

Reviewer 2: No, that is my expected.
 we should highlight `namespace_name`, `pod_name`, and `container_name`  in the example when enrichment is KubernetesMinimal

Author: done as discussed

----

.Example syslog message output with `enrichment: KubernetesMinimal`

[source, text]
----

> 2025-03-03T11:42:31+00:00 anliazm3-m8hrk-worker-westus-djq2j syslogaosqersyslogserverd846bb9b: namespace_name=syslog-aosqe, container_name=rsyslog, pod_name=rsyslogserver-d846bb9bc-l2tjr, message={"@timestamp":"2025-03-03T11:42:31.862624122Z","facility":"local0","hostname":"anliazm3-m8hrk-worker-westus-djq2j","kubernetes":{"annotations":{"k8s.ovn.org/pod-networks":"{\"default\":{\"ip_addresses\":[\"10.129.2.40/23\"],\"mac_address\":\"0a:58:0a:81:02:28\",\"gateway_ips\":[\"10.129.2.1\"],\"routes\":[{\"dest\":\"10.128.0.0/14\",\"nextHop\":\"10.129.2.1\"},{\"dest\":\"172.30.0.0/16\",\"nextHop\":\"10.129.2.1\"},{\"dest\":\"169.254.169.5/32\",\"nextHop\":\"10.129.2.1\"},{\"dest\":\"100.64.0.0/16\",\"nextHop\":\"10.129.2.1\"}],\"ip_address\":\"10.129.2.40/23\",\"gateway_ip\":\"10.129.2.1\"}}","k8s.v1.cni.cncf.io/network-status":"[{\n    \"name\": \"ovn-kubernetes\",\n    \"interface\": \"eth0\",\n    \"ips\": [\n        \"10.129.2.40\"\n    ],\n    \"mac\": \"0a:58:0a:81:02:28\",\n    \"default\": true,\n    \"dns\": {}\n}]","openshift.io/generated-by":"OpenShiftNewApp","openshift.io/scc":"privileged"},"container_id":"cri-o://699290ce5f3af668ebebd1789d5df9bfb1af57ff2bbbeb6241f56af845bbec87","container_image":"quay.io/openshifttest/rsyslogd-container@sha256:e806eb41f05d7cc6eec96bf09c7bcb692f97562d4a983cb019289bd048d9aee2","container_image_id":"quay.io/openshifttest/rsyslogd-container@sha256:821f81c0deea63c4f06481e3773eac3f55892c69c91b3751b0ba61fb6decb905","container_iostream":"stderr","container_name":"rsyslog","labels":{"appname":"rsyslogserver","component":"rsyslogserver","pod-template-hash":"d846bb9bc","provider":"aosqe"},"namespace_id":"ec872692-4bde-45d3-ae4b-89ba35628a3d","namespace_labels":{"kubernetes_io_metadata_name":"syslog-aosqe","pod-security_kubernetes_io_audit":"privileged","pod-security_kubernetes_io_audit-version":"v1.24","pod-security_kubernetes_io_warn":"privileged","pod-security_kubernetes_io_warn-version":"v1.24"},"namespace_name":"syslog-aosqe","pod_id":"54a774f8-9e87-4867-b285-ef868cca6032","pod_ip":"10.129.2.40","pod_name":"rsyslogserver-d846bb9bc-l2tjr","pod_owner":"ReplicaSet/rsyslogserver-d846bb9bc"},"level":"default","log_source":"container","log_type":"application","message":"rsyslogd: netstream session 0x7f1a5ad94620 from 10.129.0.92 will be closed due to error [v8.2102.0-3.fc34 try https://www.rsyslog.com/e/2078 ]","openshift":{"cluster_id":"4fd3bf05-1e95-46cf-8480-62fc9435079c","sequence":1741002167861044318},"proc_id":"-","severity":"informational","tag":"syslogaosqersyslogserverd846bb9b","timestamp":"2025-03-03T11:42:31.862624122Z"}

Reviewer 2: How about remove some fileds from this example?

Author: done, wdtyt?

----

[id="cluster-logging-collector-log-forward-examples-syslog-parms_{context}"]

> == Syslog parameters

Reviewer 2: We had introduce these Syslog parameters  above. I think we can drop the paragraph below

Author: done

:_mod-docs-content-type: PROCEDURE
[id="cluster-logging-collector-log-forward-syslog-6x_{context}"]
= Forwarding logs using the syslog protocol

You can use the *syslog* link:https://tools.ietf.org/html/rfc3164[RFC3164] or link:https://tools.ietf.org/html/rfc5424[RFC5424] protocol to send a copy of your logs to an external log aggregator that is configured to accept the protocol instead of, or in addition to, the default Elasticsearch log store. You are responsible for configuring the external log aggregator, such as a syslog server, to receive the logs from {product-title}.

To configure log forwarding using the *syslog* protocol, you must create a `ClusterLogForwarder` custom resource (CR) with one or more outputs to the syslog servers, and pipelines that use those outputs. The syslog output can use a UDP, TCP, or TLS connection.

.Prerequisites

* You must have a logging server that is configured to receive the logging data using the specified protocol or format.

.Procedure

> . Create or edit a YAML file that defines the `ClusterLogForwarder` CR object:

Reviewer 2: That is a 5.x example:
I suggest using a simple obsclf  example and include all appropriate properties.   You can find the properties description in  https://github.com/openshift/cluster-logging-operator/blob/master/bundle/manifests/observability.openshift.io_clusterlogforwarders.yaml#L2320

```
apiVersion: observability.openshift.io/v1
kind: ClusterLogForwarder
metadata:
  name: collector
spec:
  managementState: Managed
  outputs:
  - name: rsyslog-east
    syslog:
      appName: foo-{.bar||"none"}
      enrichment: KubernetesMinimal
      facility: local0
      msgId: foo-{.bar||"none"}
      payloadKey: '{.bar}'
      procId: foo-{.bar||"none"}
      rfc: RFC3164
      severity: informational
      tuning:
        deliveryMode: AtLeastOnce
      url: tls://rsyslogserver.syslog-aosqe.svc:6514
    tls:
      ca:
        key: ca-bundle.crt
        secretName: syslog-secret
    type: syslog
  pipelines:
  - inputRefs:
    - application
    name: syslog-east
    outputRefs:
    - rsyslog-east
  serviceAccount:
    name: logcollector

```

Reviewer 2: yes, that is OK

<7> Optional: Specify a name for the pipeline.
<8> Specify which log types to forward by using the pipeline: `application,` `infrastructure`, or `audit`.
<9> Specify the name of the output to use when forwarding logs with this pipeline.
<10> Optional: Specify the `default` output to forward logs to the internal Elasticsearch instance.
<11> Optional: String. One or more labels to add to the logs. Quote values like "true" so they are recognized as string values, not as a boolean.
<12> Optional: Configure multiple outputs to forward logs to other external log aggregators of any supported type:
** A name to describe the pipeline.
** The `inputRefs` is the log type to forward by using the pipeline: `application,` `infrastructure`, or `audit`.
** The `outputRefs` is the name of the output to use.
** Optional: String. One or more labels to add to the logs.

. Create the CR object:
+
[source,terminal]
----
$ oc create -f <filename>.yaml
----

[id="cluster-logging-collector-log-forward-examples-syslog-log-source_{context}"]
== Adding log source information to message output

> You can add `namespace_name`, `pod_name`, and `container_name` elements to the `message` field of the record by adding the `AddLogSource` field to your `ClusterLogForwarder` custom resource (CR).

Reviewer 2: AddLogSource is replaced with "enrichment: KubernetesMinimal"

[source,yaml]
----
  spec:
    outputs:
    - name: syslogout
      syslog:
        addLogSource: true
        facility: user
        payloadKey: message
        rfc: RFC3164
        severity: debug
        tag: mytag
      type: syslog
      url: tls://syslog-receiver.openshift-logging.svc:24224
    pipelines:
    - inputRefs:
      - application
      name: test-app
      outputRefs:
      - syslogout
----

[NOTE]
====
This configuration is compatible with both RFC3164 and RFC5424.
====

> .Example syslog message output without `AddLogSource`

Reviewer 2: Example: when enrichment:  None
`2025-03-03T11:48:01+00:00 anliazm3-m8hrk-worker-westus-jl26f syslogaosqersyslogserverd846bb9b: {"@timestamp":"2025-03-03T11:48:01.440861985Z","facility":"local0","hostname":"anliazm3-m8hrk-worker-westus-jl26f","kubernetes":{"annotations":{"k8s.ovn.org/pod-networks":"{\"default\":{\"ip_addresses\":[\"10.131.0.30/23\"],\"mac_address\":\"0a:58:0a:83:00:1e\",\"gateway_ips\":[\"10.131.0.1\"],\"routes\":[{\"dest\":\"10.128.0.0/14\",\"nextHop\":\"10.131.0.1\"},{\"dest\":\"172.30.0.0/16\",\"nextHop\":\"10.131.0.1\"},{\"dest\":\"169.254.169.5/32\",\"nextHop\":\"10.131.0.1\"},{\"dest\":\"100.64.0.0/16\",\"nextHop\":\"10.131.0.1\"}],\"ip_address\":\"10.131.0.30/23\",\"gateway_ip\":\"10.131.0.1\"}}","k8s.v1.cni.cncf.io/network-status":"[{\n    \"name\": \"ovn-kubernetes\",\n    \"interface\": \"eth0\",\n    \"ips\": [\n        \"10.131.0.30\"\n    ],\n    \"mac\": \"0a:58:0a:83:00:1e\",\n    \"default\": true,\n    \"dns\": {}\n}]","openshift.io/generated-by":"OpenShiftNewApp","openshift.io/scc":"privileged"},"container_id":"cri-o://f8da362708b701483429bc448851438417ad654d304cbf302c9f26625b99aeed","container_image":"quay.io/openshifttest/rsyslogd-container@sha256:e806eb41f05d7cc6eec96bf09c7bcb692f97562d4a983cb019289bd048d9aee2","container_image_id":"quay.io/openshifttest/rsyslogd-container@sha256:821f81c0deea63c4f06481e3773eac3f55892c69c91b3751b0ba61fb6decb905","container_iostream":"stderr","container_name":"rsyslog","labels":{"appname":"rsyslogserver","component":"rsyslogserver","pod-template-hash":"d846bb9bc","provider":"aosqe"},"namespace_id":"ec872692-4bde-45d3-ae4b-89ba35628a3d","namespace_labels":{"kubernetes_io_metadata_name":"syslog-aosqe","pod-security_kubernetes_io_audit":"privileged","pod-security_kubernetes_io_audit-version":"v1.24","pod-security_kubernetes_io_warn":"privileged","pod-security_kubernetes_io_warn-version":"v1.24"},"namespace_name":"syslog-aosqe","pod_id":"25ce6cc5-79bd-46d0-a92c-22d573367e12","pod_ip":"10.131.0.30","pod_name":"rsyslogserver-d846bb9bc-vn4fk","pod_owner":"ReplicaSet/rsyslogserver-d846bb9bc"},"level":"default","log_source":"container","log_type":"application","message":"rsyslogd: error during parsing file /etc/rsyslog/conf/rsyslog.conf, on or before line 22: invalid character ''' - is there an invalid escape sequence somewhere? [v8.2102.0-3.fc34 try https://www.rsyslog.com/e/2207 ]","openshift":{"cluster_id":"4fd3bf05-1e95-46cf-8480-62fc9435079c","sequence":1741002487735423758},"proc_id":"-","severity":"informational","tag":"syslogaosqersyslogserverd846bb9b","timestamp":"2025-03-03T11:48:01.440861985Z"}`

[source, text]
----
<15>1 2020-11-15T17:06:14+00:00 fluentd-9hkb4 mytag - - -  {"msgcontent"=>"Message Contents", "timestamp"=>"2020-11-15 17:06:09", "tag_key"=>"rec_tag", "index"=>56}
----

> .Example syslog message output with `AddLogSource`

Reviewer 2: Example: when enrichment: KubernetesMinimal
`2025-03-03T11:42:31+00:00 anliazm3-m8hrk-worker-westus-djq2j syslogaosqersyslogserverd846bb9b: namespace_name=syslog-aosqe, container_name=rsyslog, pod_name=rsyslogserver-d846bb9bc-l2tjr, message={"@timestamp":"2025-03-03T11:42:31.862624122Z","facility":"local0","hostname":"anliazm3-m8hrk-worker-westus-djq2j","kubernetes":{"annotations":{"k8s.ovn.org/pod-networks":"{\"default\":{\"ip_addresses\":[\"10.129.2.40/23\"],\"mac_address\":\"0a:58:0a:81:02:28\",\"gateway_ips\":[\"10.129.2.1\"],\"routes\":[{\"dest\":\"10.128.0.0/14\",\"nextHop\":\"10.129.2.1\"},{\"dest\":\"172.30.0.0/16\",\"nextHop\":\"10.129.2.1\"},{\"dest\":\"169.254.169.5/32\",\"nextHop\":\"10.129.2.1\"},{\"dest\":\"100.64.0.0/16\",\"nextHop\":\"10.129.2.1\"}],\"ip_address\":\"10.129.2.40/23\",\"gateway_ip\":\"10.129.2.1\"}}","k8s.v1.cni.cncf.io/network-status":"[{\n    \"name\": \"ovn-kubernetes\",\n    \"interface\": \"eth0\",\n    \"ips\": [\n        \"10.129.2.40\"\n    ],\n    \"mac\": \"0a:58:0a:81:02:28\",\n    \"default\": true,\n    \"dns\": {}\n}]","openshift.io/generated-by":"OpenShiftNewApp","openshift.io/scc":"privileged"},"container_id":"cri-o://699290ce5f3af668ebebd1789d5df9bfb1af57ff2bbbeb6241f56af845bbec87","container_image":"quay.io/openshifttest/rsyslogd-container@sha256:e806eb41f05d7cc6eec96bf09c7bcb692f97562d4a983cb019289bd048d9aee2","container_image_id":"quay.io/openshifttest/rsyslogd-container@sha256:821f81c0deea63c4f06481e3773eac3f55892c69c91b3751b0ba61fb6decb905","container_iostream":"stderr","container_name":"rsyslog","labels":{"appname":"rsyslogserver","component":"rsyslogserver","pod-template-hash":"d846bb9bc","provider":"aosqe"},"namespace_id":"ec872692-4bde-45d3-ae4b-89ba35628a3d","namespace_labels":{"kubernetes_io_metadata_name":"syslog-aosqe","pod-security_kubernetes_io_audit":"privileged","pod-security_kubernetes_io_audit-version":"v1.24","pod-security_kubernetes_io_warn":"privileged","pod-security_kubernetes_io_warn-version":"v1.24"},"namespace_name":"syslog-aosqe","pod_id":"54a774f8-9e87-4867-b285-ef868cca6032","pod_ip":"10.129.2.40","pod_name":"rsyslogserver-d846bb9bc-l2tjr","pod_owner":"ReplicaSet/rsyslogserver-d846bb9bc"},"level":"default","log_source":"container","log_type":"application","message":"rsyslogd: netstream session 0x7f1a5ad94620 from 10.129.0.92 will be closed due to error [v8.2102.0-3.fc34 try https://www.rsyslog.com/e/2078 ]","openshift":{"cluster_id":"4fd3bf05-1e95-46cf-8480-62fc9435079c","sequence":1741002167861044318},"proc_id":"-","severity":"informational","tag":"syslogaosqersyslogserverd846bb9b","timestamp":"2025-03-03T11:42:31.862624122Z"}
`
